### PR TITLE
Added system config selection for all users configuration

### LINKED ISSUE #6538: System-wide config for all system users
### Problem ###

The config doesn't persist across users when you configure for one user, and then switch to another. The config is not shared between users; each user has to have their own config. In other words, once a config has been created, when a new user logs into the same computer, the config is essentially lost.

### Solution ###

Add an option in the Settings dialog to save the config for all users. The tricky part is making that work on all platforms without affecting the default single user config behavior. Command-line arguments should still override any settings files, and user config should override the global config.

### PR DESCRIPTION
**Pull request by Jnewbon**
Fixes #6538

### CODE REVIEW

File: src/gui/src/ConfigWriter.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
+//
```
Comment (nbolton) [inline src/gui/src/ConfigWriter.cpp:1]: Missing standard heading comment.

File: src/gui/src/ConfigWriter.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+#include "ConfigBase.h"
+
+namespace GUI {
+    namespace Config {
+        //Assignment of static variable
+        ConfigWriter *ConfigWriter::s_pConfiguration = nullptr;
+
+
+        ConfigWriter *ConfigWriter::make() {
+            // Only one ConfigWriter can exist at any one time (Singolton)
```
Comment (nbolton) [inline src/gui/src/ConfigWriter.cpp:21]: Typo: singleton

File: src/gui/src/main.cpp [hunk 3]
```diff
@@ -88,9 +88,10 @@ int main(int argc, char* argv[])
 #ifndef Q_OS_WIN
     QApplication::setQuitOnLastWindowClosed(false);
 #endif
+    //S
```
Comment (nbolton) [inline src/gui/src/main.cpp:91]: Rogue comment?

File: src/gui/src/ConfigWriter.cpp (hunk 1 above)
Comment (Jnewbon) [reply src/gui/src/ConfigWriter.cpp]: Added header with correct dates

File: src/gui/src/ConfigWriter.cpp (hunk 2 above)
Comment (Jnewbon) [reply src/gui/src/ConfigWriter.cpp]: Corrected

File: src/gui/src/main.cpp (hunk 3 above)
Comment (Jnewbon) [reply src/gui/src/main.cpp]: That's where the S went... :)

**Pull request merged**